tools/clang/unittests/SPIRV/FileTestFixture.cpp

```diff
@@ -0,0 +1,98 @@
+//===- FileTestFixture.cpp ------------- File Test Fixture Implementation -===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "FileTestFixture.h"
+
+#include <fstream>
+
+#include "FileTestUtils.h"
+#include "effcee/effcee.h"
+
+namespace clang {
+namespace spirv {
+
+bool FileTest::parseInputFile() {
+  const char hlslStartLabel[] = "// Run:";
+  bool foundRunCommand = false;
+  std::ostringstream checkCommandStream;
+  std::ifstream inputFile;
+  inputFile.exceptions(std::ifstream::failbit);
+
+  try {
+    inputFile.open(inputFilePath);
+    for (std::string line; std::getline(inputFile, line);) {
+      if (line.find(hlslStartLabel) != std::string::npos) {
```

Comment:
This can be further simplified actually. We can read in the whole file together and feed it to effcee as check commands. By searching the`run` directive and create a `string` out of that particular line we can avoid parsing all the lines and creating `string`s for them. But I won't hold back this PR. Will create a task for this.